runtime/rasdump/dmpagent.c

```diff
@@ -437,7 +437,7 @@ static const J9RASdumpSpec rasDumpSpecs[] =
 		  NULL,
 		  NULL,
 		  0,
-		  J9RAS_DUMP_DO_SUSPEND_OTHER_DUMPS,
+		  J9RAS_DUMP_DO_EXCLUSIVE_VM_ACCESS,
```

Comment:
What was causing the deadlock that this needs to acquire exclusive?  Shouldn't `J9RAS_DUMP_DO_SUSPEND_OTHER_DUMPS` be sufficient?

It would be easier to approve this if you detailed the deadlock in the PR description and how this fixed it.

Reply:
I've added some additional details to the description.

Reply:
I'm beginning to think the problem is more general: that `J9RAS_DUMP_DO_SUSPEND_OTHER_DUMPS` and `J9RAS_DUMP_DO_EXCLUSIVE_VM_ACCESS` don't play well together. The thread handling an agent with `J9RAS_DUMP_DO_SUSPEND_OTHER_DUMPS` isn't waiting in a way that will permit other agents that `J9RAS_DUMP_DO_EXCLUSIVE_VM_ACCESS` to proceed. If we get past that, we may also need for threads to recognize that the VM has exited.

Reply:
@theresa-m can you take a look at how `J9RAS_DUMP_DO_SUSPEND_OTHER_DUMPS` & `J9RAS_DUMP_DO_EXCLUSIVE_VM_ACCESS` interact and see if there's a straightforward fix here?

Reply:
yes I will take a look.

as a separate question why is J9RAS_DUMP_DO_SUSPEND_OTHER_DUMPS is preferable for the exit agent over J9RAS_DUMP_DO_EXCLUSIVE_VM_ACCESS?

Reply:
Exclusive access is to be a big hammer that we try not swing.  It's usually better to find another way to solve a problem than to take exclusive.

If there's a reason why exclusive is the right answer here, we can discuss further.